imagebuildah/stage_executor.go

```diff
@@ -446,6 +500,63 @@ func (s *StageExecutor) runStageMountPoints(mountList []string) (map[string]inte
 					if fromErr != nil {
 						return nil, errors.Wrapf(fromErr, "unable to resolve argument %q", kv[1])
 					}
+					// If additional buildContext contains this
+					// give priority to that and break if additional
+					// is not an external image.
+					if additionalBuildContext, ok := s.executor.additionalBuildContexts[from]; ok {
+						if additionalBuildContext != nil {
+							if additionalBuildContext.IsImage {
+								mountPoint, err := s.getImageRootfs(s.ctx, additionalBuildContext.Value)
+								if err != nil {
+									return nil, errors.Errorf("%s from=%s: no stage or image found with that name", flag, from)
+								}
+								// The `from` in stageMountPoints should point
+								// to `mountPoint` replaced from additional
+								// build-context. Reason: Parser will use this
+								//  `from` to refer from stageMountPoints map later.
+								stageMountPoints[from] = internal.StageMountDetails{IsStage: false, MountPoint: mountPoint}
+								break
+							} else {
+								// Most likely this points to path on filesystem
+								// or external tar archive, Treat it as a stage
+								// nothing is different for this. So process and
+								// point mountPoint to path on host and it will
+								// be automatically handled correctly by since
+								// GetBindMount will honor IsStage:false while
+								// processing stageMountPoints.
+								mountPoint := additionalBuildContext.Value
+								if additionalBuildContext.IsURL {
+									// Check if following buildContext was already
+									// downloaded before in any other RUN step. If yes
+									// lets check if we are hitting any cache and cache
+									// actually exists on host.
+									cacheExistsOnHost := false
+									if additionalBuildContext.DownloadedCache != "" {
+										if stat, err := os.Stat(additionalBuildContext.DownloadedCache); err == nil && stat.IsDir() {
+											cacheExistsOnHost = true
+										}
+									}
+									if !cacheExistsOnHost {
```

Comment:
It looks like `cacheExistsOnHost` can be eliminated, since `DownloadedCache` is only ever set after the directory is created and populated.